### PR TITLE
Move Core logic from View to Controller for HomePage

### PR DESCRIPTION
**Pull request by PlasmaIntec**
### Please check if the PR fulfills these requirements:

* [ ] The PR title and message follows our [guidelines](https://submitty.org/developer/how_to_contribute)
* [ ] Tests for the changes have been added/updated (if possible)
* [ ] Documentation has been updated/added if relevant

### CODE REVIEW

File: .setup/INSTALL_SUBMITTY_HELPER_SITE.sh [hunk 1]
```diff
@@ -4,11 +4,15 @@
 ################################################################################################################
 # COPY THE 1.0 Grading Website
 
+start=$SECONDS
```
Comment (dakriy) [inline .setup/INSTALL_SUBMITTY_HELPER_SITE.sh]: What does any of this have to do with moving core logic from view to controller for the home page? I think a change this big should have it's own issue and PR.

File: site/app/controllers/HomePageController.php [hunk 2]
```diff
@@ -135,21 +136,64 @@ function(Course $course) {
...
+        $archived_courses = $courses["data"]["archived_courses"];
+        $unarchived_courses = $courses["data"]["unarchived_courses"];
+        $course_types = [$unarchived_courses, $archived_courses];
+        $rank_titles = [
+            User::GROUP_INSTRUCTOR              => "Instructor:",
+            User::GROUP_FULL_ACCESS_GRADER      => "Full Access Grader:",
+            User::GROUP_LIMITED_ACCESS_GRADER   => "Grader:",
+            User::GROUP_STUDENT                 => "Student:"
+        ];
+        $statuses = array();
```
Comment (dakriy) [inline site/app/controllers/HomePageController.php]: This and things like it can be simplified to
`$statuses = [];` which is the prefered way in PHP7

File: site/app/controllers/HomePageController.php [hunk 3]
```diff
@@ -135,21 +136,64 @@ function(Course $course) {
...
+
+            //Create rank lists
+            for ($i = 1; $i < 5; $i++){
+                $ranks[$i] = [];
+                $ranks[$i]["title"] = $rank_titles[$i];
+                $ranks[$i]["courses"] = [];
+            }
+
+            foreach ($course_type as $course) {
+                array_push($ranks[$course["user_group"]]["courses"], $course);
```
Comment (dakriy) [inline site/app/controllers/HomePageController.php]: What's wrong with `$ranks[$course["user_group"]]["courses"][] = $course`?

File: site/app/controllers/HomePageController.php [hunk 4]
```diff
@@ -135,21 +136,64 @@ function(Course $course) {
...
+                $user,
+                $statuses,
                 $this->core->getConfig()->getUsernameChangeText()
             )
         );
     }
 
+    private function appendRank($courses, $user) {
+        //Assemble courses into rank lists
+        $ranks = array();
```
Comment (dakriy) [inline site/app/controllers/HomePageController.php]: `$ranks = []`?

File: site/app/controllers/HomePageController.php [hunk 5]
```diff
@@ -135,21 +136,64 @@ function(Course $course) {
...
+            User::GROUP_INSTRUCTOR              => "Instructor:",
+            User::GROUP_FULL_ACCESS_GRADER      => "Full Access Grader:",
+            User::GROUP_LIMITED_ACCESS_GRADER   => "Grader:",
+            User::GROUP_STUDENT                 => "Student:"
+        ];
+        $statuses = array();
+
+        foreach($course_types as $course_type) {
+
+            $ranks = array();
```
Comment (dakriy) [inline site/app/controllers/HomePageController.php]: `$ranks = [];`?

File: site/app/libraries/database/DatabaseQueries.php [hunk 6]
```diff
@@ -2494,6 +2494,31 @@ public function getUnarchivedCoursesById($user_id) {
...
+         CASE WHEN SUBSTRING(u.semester, 2, 2) ~ '\\d+' THEN SUBSTRING(u.semester, 2, 2)::INT
+              ELSE 0
+         END DESC,
+         CASE WHEN SUBSTRING(u.semester, 1, 1) = 's' THEN 2
+              WHEN SUBSTRING(u.semester, 1, 1) = 'u' THEN 3
+              WHEN SUBSTRING(u.semester, 1, 1) = 'f' THEN 4
+              ELSE 1
+         END DESC,
+         u.course ASC", array($user_id));
+        $return = array();
```
Comment (dakriy) [inline site/app/libraries/database/DatabaseQueries.php]: `$return = [];`?

File: site/app/libraries/database/DatabaseQueries.php [hunk 7]
```diff
@@ -2494,6 +2494,31 @@ public function getUnarchivedCoursesById($user_id) {
...
+ORDER BY u.user_group ASC,
+         CASE WHEN SUBSTRING(u.semester, 2, 2) ~ '\\d+' THEN SUBSTRING(u.semester, 2, 2)::INT
+              ELSE 0
+         END DESC,
+         CASE WHEN SUBSTRING(u.semester, 1, 1) = 's' THEN 2
+              WHEN SUBSTRING(u.semester, 1, 1) = 'u' THEN 3
+              WHEN SUBSTRING(u.semester, 1, 1) = 'f' THEN 4
+              ELSE 1
+         END DESC,
+         u.course ASC", array($user_id));
```
Comment (dakriy) [inline site/app/libraries/database/DatabaseQueries.php]: `[$user_id]`?

File: site/app/libraries/database/DatabaseQueries.php [hunk 8]
```diff
@@ -2530,6 +2555,31 @@ public function getArchivedCoursesById($user_id) {
...
+        $return = array();
```
Comment (dakriy) [inline site/app/libraries/database/DatabaseQueries.php]: Here too I guess lol

File: site/app/views/HomePageView.php [hunk 9]
```diff
@@ -12,40 +12,7 @@ class HomePageView extends AbstractView {
...
-
-            //Filter any ranks with no courses
-            $ranks = array_filter($ranks, function($rank) {
-                return count($rank["courses"]) > 0;
-            });
-            $statuses[] = $ranks;
-        }
-
-
+    public function showHomePage(User $user, $statuses = array(), $change_name_text) {
```
Comment (dakriy) [inline site/app/views/HomePageView.php]: If it's gotta be an array might as well type it. And it looks better if you use [] again. So:
`public function showHomePage(User $user, array $statuses = [], $change_name_text)`
Not sure what type $change_name_text is supposed to be

File: watch.sh [hunk 10]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+
+vagrant ssh -c "python3 /usr/local/submitty/GIT_CHECKOUT/Submitty/.setup/bin/code_watcher.py"
```
Comment (dakriy) [inline watch.sh]: This is cool and all but it has nothing to do with the issue you are solving. :/ Make separate PR for it?

**Comment by CalebJ2** [comment]
@PlasmaIntec What's the status of this? Any more changes coming?
